### ISSUE TITLE
Improved hints/autocomplete for union literal types

### ISSUE THREAD

**Issue by alisd23** [Microsoft/vscode#45269]
Currently, the hints for union literal type parameters is OK, but it could be better.
Given this example union string literal type (taken directly from the typescript docs: https://basarat.gitbooks.io/typescript/docs/types/literal-types.html):

```ts
type CardinalDirection = 'North' | 'East' | 'South' | 'West';

function move(distance: number, direction: CardinalDirection) {
    // ...
}
```
> **Note**
> For the following, I'm not sure about the terminology you guys use for the hints, but when I say hints I mean the list that displays after pressing `.` after objects/class instances, which can be navigated with keyboard controls and selected by clicking or pressing enter.

It would be nice to display this selectable list of possible values for the `direction` parameter at the point of typing `move(1, ':cursor:')`. Currently this does *kind of* work, where the hint list shows only after pressing a trigger character like `.`, which is kind of hacky and probably hard to notice unless it is explicitly explained to the user.

I know that some libraries utilize union literal string types for their API's (with much bigger sets of possible string literals), so it would great to display possible literal values for these.

The most recent example I have seen is for the end-to-end testing library [Cypress](https://docs.cypress.io/api/commands/should.html#Syntax).

Thanks

>**Edit**
> I guess this would also apply to `boolean` and `number` literals, and mixtures of these (which may be more of a stretch).

**Comment by mjbvz** [comment]
Duplicate of #25034

You can also trigger suggestions manually after the `'` by pressing ctrl+space
